### DISCUSSION TITLE
Use original commit message as PR title

### DISCUSSION THREAD

**Discussion by Nef10** [BetaHuhn/repo-file-sync-action#103] [Ideas] [answered]
When using the `ORIGINAL_MESSAGE` option, I would prefer to have the first line of the commit message as the PR title. In my workflow this would allow me to automatically have a nice change log, instead of what it is currently producing: 

<img width="1121" alt="Screen Shot 2021-09-06 at 1 42 09 AM" src="https://user-images.githubusercontent.com/4551135/132187669-e09e9410-a43f-42d5-ab28-944f016e9175.png">

Wondering if this is something which is useful for others as well? And I am also not sure how that should be implemented, should it be a separate flag, or should it just be part of `ORIGINAL_MESSAGE`?

Overall I think it comes back to how the action is used. I like to keep the action more in the background, so that each repo looks standalone, and one does not necessary notice at a first glance that this action is running in the background to help. For  this the commit message is the most important, as this is was shows up on in the repo itself. But the PR title would improve this even more.

**Comment by BetaHuhn** [comment, marked as answer]
We could add another option, maybe `COMMIT_AS_PR_TITLE`? When enabled and the PR only includes one commit it will use its message as the PR title instead of the default one.

What do you think?

**Reply by Nef10** [reply]
I think that sounds good. When I get some time I'll work on it :)

**Reply by BetaHuhn** [reply]
Thanks for the [PR](https://github.com/BetaHuhn/repo-file-sync-action/pull/115)!

With the `ORIGINAL_MESSAGE` and `COMMIT_AS_PR_TITLE` option it gets a bit confusing for new users, I think it would be good to add an example/explanation to the README's [examples](https://github.com/BetaHuhn/repo-file-sync-action#-examples) section. What do you think?

**Reply by Nef10** [reply]
Yeah, we could add a some more explanation to it - I don't think an example is really necessary as the most the user can to is set two values to true.

**Reply by BetaHuhn** [reply]
Yeah true. It doesn't really fit in the action inputs table though, maybe we need to add another sub section like `### Customizing the PR and commits` which explains all of the options available to change a PR and commit messages.
